plugins/chat/app/jobs/regular/chat_notify_mentioned.rb

```diff
@@ -111,11 +111,9 @@ def create_notification!(membership, notification_data)
           data: notification_data.to_json,
           read: is_read,
         )
-      ChatMention.create!(
-        notification: notification,
-        user: membership.user,
-        chat_message: @chat_message,
-      )
+
+      mention = ChatMention.find_by(user: membership.user, chat_message: @chat_message)
```

Comment:
I guess this could _potentially_ be nil as well so we should handle that

Reply:
I was keeping an eye on it when developing, and I wasn't able to reproduce a situation when there is no mention here. In general, I'd expect this to work fine, but this job and the code that calls it is pretty complicated, so we can't be 100% sure.

But the question is how to handle it 🤔.

This job is for notifying a user about a mention. So if we're inside the job, and we discover that there is no mention, that would mean that something went wrong in the code that called the job. Because if there is no mention, then there is nothing to notify about and the job shouldn't be called at all. If we do this:

```ruby
mention.update!(notification: notification) if mention.present?
```

we won't actually handle the problem, in fact we'll silently swallow it. Also, in this case, since the notification won't be associated with a mention, we may have other exceptions with this notification later. It's easier to raise an exception earlier, that will be simpler to investigate.

What we can do is we can reraise more explicit exception here:

```ruby
raise "Trying to notify about a mention, but the mention wasn't found" if mention.blank?
```

What do you think?

Reply:
> we won't actually handle the problem, in fact we'll silently swallow it. Also, in this case, since the notification won't be associated with a mention, we may have other exceptions with this notification later. It's easier to raise an exception earlier, that will be simpler to investigate.

IMO just try to find the mention before creating the notification, and if it does not exist just early return. As you say it shouldn't really happen, but then at least we aren't creating the notification record for no reason

Reply:
Let's do that. The one more thing is that we don't need to also send notifications in this case, so I checked if a mention is `nil` even earlier.